os-server/src/main/java/app/openschool/course/module/quiz/QuizServiceImpl.java

```diff
@@ -0,0 +1,202 @@
+      Locale locale) {
+    AtomicInteger rightAnswers =
+        checkQuestionsAndReturnRightAnswersCount(
+            enrolledQuiz.getQuiz(), enrolledQuizAssessmentRequestDto);
+    String assessmentResult =
+        getAssessmentResult(rightAnswers.get(), enrolledQuiz.getQuiz(), locale);
+    updateEnrolledQuiz(rightAnswers.get(), enrolledQuiz);
+    return EnrolledQuizAssessmentResponseMapper.toEnrolledQuizAssessmentResponseDto(
+        rightAnswers.get(),
+        assessmentResult,
+        enrolledQuiz.getQuiz().getPassingScore(),
+        enrolledQuiz.getQuiz().getMaxGrade());
+  }
+
+  private void updateEnrolledQuiz(int studentGrade, EnrolledQuiz enrolledQuiz) {
+    enrolledQuiz.setStudentGrade(studentGrade);
+    if (studentGrade >= enrolledQuiz.getQuiz().getPassingScore()) {
+      enrolledQuiz.setQuizStatus(QuizStatus.isCompleted());
+    } else {
+      enrolledQuiz.setQuizStatus(QuizStatus.isFailed());
+    }
+    enrolledQuizRepository.save(enrolledQuiz);
+  }
+
+  private String getAssessmentResult(int rightAnswers, Quiz quiz, Locale locale) {
+    return quiz.getPassingScore() <= rightAnswers
+        ? messageSource.getMessage("quiz.completed", null, locale)
+        : messageSource.getMessage("quiz.failed", null, locale);
+  }
+
+  private AtomicInteger checkQuestionsAndReturnRightAnswersCount(
+      Quiz quiz, EnrolledQuizAssessmentRequestDto enrolledQuizAssessmentRequestDto) {
+    AtomicInteger rightAnswers = new AtomicInteger(0);
+    quiz.getQuestions()
+        .forEach(
+            question -> {
+              if (checkQuestion(question, enrolledQuizAssessmentRequestDto)) {
+                rightAnswers.getAndIncrement();
+              }
+            });
+    return rightAnswers;
+  }
+
+  private boolean checkQuestion(
+      Question question, EnrolledQuizAssessmentRequestDto enrolledQuizAssessmentRequestDto) {
+    AtomicBoolean isRightAnswer = new AtomicBoolean(false);
+    enrolledQuizAssessmentRequestDto
+        .getQuestionWithChosenAnswerDtoSet()
+        .forEach(
+            questionWithChosenAnswerDto -> {
+              if (Objects.equals(questionWithChosenAnswerDto.getQuestionId(), question.getId())) {
+                isRightAnswer.set(
+                    checkQuestionAnswers(
+                        question, questionWithChosenAnswerDto.getChosenAnswersIds()));
+              }
+            });
+    return isRightAnswer.get();
+  }
+
+  private boolean checkQuestionAnswers(Question question, List<Long> chosenAnswerIds) {
+    AtomicInteger rightAnswers = new AtomicInteger();
+    question
+        .getAnswerOptions()
+        .forEach(
+            answerOption -> {
+              if (chosenAnswerIds.contains(answerOption.getId()) && answerOption.isRightAnswer()) {
+                rightAnswers.getAndIncrement();
+              }
+            });
+    return rightAnswers.get() == question.getRightAnswersCount();
+  }
+
+  private void checkIfTheQuizBelongsToCurrentMentor(Quiz quiz) {
```

Comment:
we should select quizzes that belongs to mentor so we will not need to make those validations.